src/iDynFor/iDynTreePinocchioConversions.h

```diff
@@ -4,14 +4,143 @@
+      return *this;
+    }
+    std::ostream* log;
+};
+
+template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+class iDynTreeVisitor : public iDynTreeVisitorBaseTpl<Scalar, Options>
+{
+  public:
+    typedef iDynTreeVisitorBaseTpl<Scalar, Options> Base;
+    typedef typename Base::JointType      JointType;
+    typedef typename Base::Vector3        Vector3;
+    typedef typename Base::VectorConstRef VectorConstRef;
+    typedef typename Base::SE3            SE3;
+    typedef typename Base::Inertia        Inertia;
+    typedef pinocchio::ModelTpl<Scalar,Options,JointCollectionTpl> Model;
+    Model& model;
+    iDynTreeVisitor (Model& model) : model(model) {}
+    void setName (const std::string& name)
+    {
+      model.name = name;
+    }
+    virtual void addRootJoint(const Inertia& Y, const std::string & body_name)
+    {
+        addFixedJointAndBody(0, SE3::Identity(), "root_joint", Y, body_name);
+        // TODO: change for the correct behavior, see 
+        //   https://github.com/stack-of-tasks/pinocchio/pull/1102 for discussions on the topic
+        //   and https://github.com/stack-of-tasks/pinocchio/blob/280005c8d99c2485ee942b6a38c4dc0edf75c706/src/parsers/urdf/model.hxx#L117
+        // appendBodyToJoint(0,Y,SE3::Identity(),body_name); 
+    }
+    virtual void addFixedJointAndBody(
+        const pinocchio::FrameIndex & parent_frame_id,
+        const pinocchio::SE3 & joint_placement,
+        const std::string & joint_name,
+        const pinocchio::Inertia & Y,
+        const std::string & body_name)
+    {
+        const pinocchio::Frame & parent_frame = model.frames[parent_frame_id];
+        const pinocchio::JointIndex parent_frame_parent = parent_frame.parent;
+
+        const pinocchio::SE3 placement = parent_frame.placement * joint_placement;
+        pinocchio::FrameIndex fid = model.addFrame(pinocchio::Frame(joint_name, parent_frame.parent, parent_frame_id,
+                                            placement, pinocchio::FIXED_JOINT, Y));
+
+        model.addBodyFrame(body_name, parent_frame_parent, placement, (int)fid);
+    }
+
+};
+
+typedef iDynTreeVisitorBaseTpl<double, 0> iDynTreeVisitorBase;
+}
+
 
+/**
+ * Convert an iDynTree::SpatialInertia to a pinocchio::Inertia
+ */
 pinocchio::Inertia toPinocchio(const iDynTree::SpatialInertia& inertiaIDynTree);
 
+/**
+ * \brief Build the pinocchio model from a iDynTree::Model .
+ *
+ * \param[in] model The iDynTree::Model to load.
+ * \param[in] verbose Print parsing info.
+ * \param[out] model Reference model where to put the parsed information.
+ * \return Return the reference on argument model for convenience.
+ *
+ * The signature of this function is inspired from the pinocchio::buildModel function
+ */
+template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+pinocchio::ModelTpl<Scalar,Options,JointCollectionTpl> &
+buildModelfromiDynTree(const iDynTree::Model & modelIDynTree,
```

Comment:
Nitpick, is it worth specifying that the output model is the ``pinocchio`` one? I guess that from the interface point of view we should define what is considered "default". I.e., is ``iDynfor`` a way to use ``pinocchio`` from ``iDynTree``, or a tool to switch between the two of them or even a library that will define its own ``Model``?

Reply:
The rationale for this is that I just appended `fromiDynTree` to the `buildModel` function as defined in https://github.com/stack-of-tasks/pinocchio/blob/a1df23c2f183d84febdc2099e5fbfdbd1fc8018b/src/parsers/urdf.hpp#L65 . If there is any strong opinion on this, I would be happy to call this `buildPinocchioModelfromiDynTree` or something similar, otherwise let's keep it as it is.

Reply:
I don't have a strong opinion, it was just a comment to understand how to define a guideline for future method names.

As discussed F2F, one possible case where it could be not clear which model we are outputting is
```cpp
auto model = buildModelfromiDynTree(..);
```
unless we make clear that by ``Model`` we always indicate a ``pinocchio`` model.

Reply:
Ack, let's change it to `buildPinocchioModelfromiDynTree` .